presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -625,6 +629,18 @@ private static OrcPredicate toOrcPredicate(TupleDomain<Subfield> domainPredicate
             return filterFunctions.build();
         }
 
+        DynamicFilterExtractResult extractDynamicFilterResult = extractDynamicFilters(filter);
+
+        // Extract the static ones
+        // TODO: james, and() is a short cut hack
```

Comment:
Remove the comment; just leave the following line. Add a comment: "dynamic filter will be added through subfield pushdown"